velox/common/memory/MappedMemory.h

```diff
@@ -147,20 +157,70 @@ class MappedMemory {
 
     // Returns the run number and the position within the run
     // corresponding to 'offset' from the start of 'this'.
-    void findRun(uint64_t offset, int32_t* index, int32_t* offsetInRun);
+    void findRun(
+        uint64_t offset,
+        int32_t* FOLLY_NONNULL index,
+        int32_t* FOLLY_NONNULL offsetInRun);
 
    private:
-    MappedMemory* mappedMemory_;
+    MappedMemory* FOLLY_NONNULL mappedMemory_;
     std::shared_ptr<MappedMemory> mappedMemoryPtr_;
     std::vector<PageRun> runs_;
     int32_t numPages_ = 0;
   };
 
+  // Represents a mmap'd run of contiguous pages that do not belong to
+  // any size class but are still accounted by the owning
+  // MappedMemory.
+  class ContiguousAllocation {
+   public:
+    ContiguousAllocation() = default;
+    ~ContiguousAllocation() {
+      if (data_ && mappedMemory_) {
+        mappedMemory_->freeContiguous(*this);
+      }
+      data_ = nullptr;
+    }
+
+    MappedMemory* FOLLY_NULLABLE mappedMemory() const {
+      return mappedMemory_;
+    }
+
+    MachinePageCount numPages() const;
+
+    template <typename T = uint8_t>
+    T* FOLLY_NULLABLE data() const {
+      return reinterpret_cast<T*>(data_);
+    }
+
+    uint64_t size() const {
```

Comment:
Is this size in bytes or number of pages? If number of pages, would you rename to numPages to clarify?